src/ui/ViewRegistry.js

```diff
@@ -52,10 +54,22 @@ define([], function () {
      * @memberof module:openmct.ViewRegistry#
      */
     ViewRegistry.prototype.addProvider = function (provider) {
+        provider._vpid = this._next_id++;
```

Comment:
Maybe view providers should just have keys? Or is the vpid just internal wiring for compatibility with legacy views? My jetlag addled brain is having trouble understanding vpid

Reply:
per discussion, yes view providers should have keys.